de.dlr.sc.virsat.server.test/src/de/dlr/sc/virsat/server/data/VirSatGitAccessTest.java

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.server.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.CoreMatchers.hasItem;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.NoHeadException;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.server.Activator;
+
+public class VirSatGitAccessTest {
+
+	private static File pathToTempUpstreamRepository;
+	private static File pathToTempLocalRepository1;
+	private static File pathToTempLocalRepository2;
+	private static Git gitUpstream;
+	
+	@Before
+	public void setUp() throws Exception {
+		pathToTempUpstreamRepository = File.createTempFile("VirSatUpstreamRepo_", "");
+		pathToTempLocalRepository1 = File.createTempFile("VirSatLocalRepo1_", "");
+		pathToTempLocalRepository2 = File.createTempFile("VirSatLocalRepo2_", "");
+		Files.delete(pathToTempUpstreamRepository.toPath());
+		Files.delete(pathToTempLocalRepository1.toPath());
+		Files.delete(pathToTempLocalRepository2.toPath());
+		Files.createDirectory(pathToTempLocalRepository1.toPath());
+		Files.createDirectory(pathToTempLocalRepository2.toPath());
+
+		gitUpstream = Git.init().setDirectory(pathToTempUpstreamRepository).call();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		try {
+			FileUtils.deleteDirectory(pathToTempUpstreamRepository);
+			FileUtils.deleteDirectory(pathToTempLocalRepository1);
+			FileUtils.deleteDirectory(pathToTempLocalRepository2);
+		} catch (Exception e) {
+			Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "Failed to remove temp directory in test" + e.getMessage()));
+		}
+	}
+
+	@Test
+	public void testCloneRepository() {

```

Comment:
I would test only one case per method, because it is then easier to spot what exactly broke when single tests start failing. It is for example much more likely that the error message changes for which the negative case checks than that something fundametnal breaks - and it is easier to spot if the positive case still works.
(But that's personal preference.)

Reply:
It is probably a good idea. would be great if we could create an example together. I leave this one as it is for the moment.